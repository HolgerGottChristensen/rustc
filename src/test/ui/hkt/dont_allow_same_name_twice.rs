// HKT test
// check-fail

//fn test<I<%J, %J>>() {}

fn test<J, J>() {} //~ ERROR: the name `J` is already used for a generic parameter in this item's generic parameters

fn main() {}
