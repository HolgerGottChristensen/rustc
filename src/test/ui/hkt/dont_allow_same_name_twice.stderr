error[E0403]: the name `J` is already used for a generic parameter in this item's generic parameters
  --> $DIR/dont_allow_same_name_twice.rs:6:12
   |
LL | fn test<J, J>() {}
   |         -  ^ already used
   |         |
   |         first use of `J`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0403`.
